fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2607,23 +2616,43 @@ private boolean areAllRecordTypesSince(@Nullable Collection<RecordType> recordTy
     }
 
     @Nonnull
-    @SuppressWarnings("PMD.EmptyCatchBlock")
-    protected CompletableFuture<Long> getRecordCountForRebuildIndexes(boolean newStore, boolean rebuildRecordCounts,
-                                                                      @Nonnull Map<Index, List<RecordType>> indexes) {
-        if (recordStoreState == null) {
-            return preloadRecordStoreStateAsync().thenCompose(vignore -> getRecordCountForRebuildIndexes(newStore, rebuildRecordCounts, indexes));
-        }
+    protected CompletableFuture<Pair<Long, Boolean>> getRecordCountForRebuildIndexes(boolean newStore, boolean rebuildRecordCounts,
```

Comment:
It seems like this method and the other one that returns a `Pair` should document what the right element in the pair is.

Reply:
Added Javadocs.